File: examples/c++/src/exception.cpp
#include <stdexcept>
#include <string>
#include "exception.hpp"

using string = std::string;

namespace Appwrite {

AppwriteException::AppwriteException(string message, int code, string response):
    runtime_error(message) {
    this->message = message;
    this->code = code;
    this->response = response;
}

const char* AppwriteException::what() const throw() {
    return this->message.c_str();
}

string AppwriteException::getResponse() {
    return this->response;
}
} // namespace Appwrite

File: examples/dotnet/src/Appwrite/Models/Exception.cs
using System;

namespace Appwrite
{
    public class AppwriteException : Exception
    {
        public int? Code;
        public string Response = null;
        public AppwriteException(string message = null, int? code = null, string response = null)
        : base(message)
        {
            this.Code = code;
            this.Response = response;
        }
        public AppwriteException(string message, Exception inner)
        : base(message, inner)
        {
        }
    }
}

